pilot/pkg/xds/ads.go

```diff
@@ -650,6 +653,7 @@ func (s *DiscoveryServer) computeProxyState(proxy *model.Proxy, request *model.P
 	if gateway && proxy.Type == model.Router {
 		proxy.SetGatewaysForProxy(push)
 	}
+	proxy.LastPushContext = push
```

Comment:
Can you move this line up?

	if push == nil {
		push = s.globalPushContext()
                proxy.LastPushContext = push
	}